### ISSUE TITLE
Can't kill python Qt Window, need to restart kernal to re-run the program.

### ISSUE THREAD

**Issue by mashtock** [spyder-ide/spyder#9991]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

I have created a ui gui using qt creator. I load the gui using the LoadUI function which is a part of PyQt5 uic Package.  Everything works find until I want to close the gui. I close the GUI by pressing the close button (the X in the top right of the window), and after that the console is stuck, and inorder to give it more commands or to re run the program I have to restart the kernal. Included here a minimal reproducable code to recreate the problem. The bug repeats itself in windows 7 and linux mint operating system. I have also tried everything in the How to Run PyQT in spyder guide to no avail.

### What steps reproduce the problem?

1. Run the code in Bug_Example.UI
2. Close the window by the close button
3. You can see that the kernel is stuck
[QT_Creator.zip](https://github.com/spyder-ide/spyder/files/3485936/QT_Creator.zip)

### What is the expected output? What do you see instead?
The expected output is that the Qt window would be closed, and I'll be able to rerun the program without the need to restart the kernel.

### Paste Traceback/Error Below (if applicable)

runfile('/home/guy/Dropbox/Python/QT_Creator/Bug_Example.py', wdir='/home/guy/Dropbox/Python/QT_Creator')
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2

![image](https://user-images.githubusercontent.com/36675372/62774457-65b54d80-baad-11e9-8370-f497957991b5.png)

## Versions

* Spyder version: 3.3.5
* Python version: 3.7.3
* Qt version: 
* PyQt version: 5
* Operating System name/version: Linux Mint/Windows 7

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by ccordoba12** [comment]
Thanks for reporting. We'll take a look at this in a future release.

**Comment by ccordoba12** [comment]
Note to self: this seems to be solved in https://stackoverflow.com/a/58537032/438386

**Comment by ccordoba12** [comment]
This should be fixed now in Spyder **4.1.1**. If that's not the case, please post a comment here.

**Comment by mg3146** [comment]
I am having this same issue trying to run this (my first time using Qt)

```
class MainWindow(QDialog):
    def __init__(self):
        super().__init__()
        
        self.setWindowTitle('Schedule Generator')
        self.resize(400,300)
        
        
        dialog_button_box = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
        dialog_button_box.accepted.connect(self.accept_clicked)
        dialog_button_box.rejected.connect(self.reject_clicked)
        dialog_button_box.sizePolicy
        
        
        text_input = QPlainTextEdit()      
        
        
        layout = QGridLayout()
        layout.addWidget(dialog_button_box, 0, 2)
        layout.addWidget(text_input, 1, 0, 2, 1)      
            
      
        
        self.setLayout(layout)
        
    def accept_clicked(self):
        print('clicked')
    
    def reject_clicked(self):
        self.reject()
        self.close()
        
if __name__ == "__main__":
    
    
    app = QApplication(sys.argv)
    window = MainWindow()
    window.show()
    sys.exit(app.exec_())
        

```
